Tools/lib/python/eupath/BigwigFilesEupathExporter.py

```diff
@@ -1,68 +1,83 @@
+     if file extension is .bw, no change
+     if file extension is .bigwig, replace with .bw
+     otherwise append .bw
+
+  dependency info:
+   - reference genome and version (as specified in dedicated ref genome parameter)
+    """
 
     # Constants
     TYPE = "BigwigFiles"
     VERSION = "1.0"
 
-    def __init__(self, args):
-        EupathExporter.Export.__init__(self,
-                                       BigwigFilesExport.TYPE,
-                                       BigwigFilesExport.VERSION,
-                                       None,
-                                       args)
 
-        # beyond the standard 7 params, this exporter requires one or more pairs of args: dataset1 dataset1.refGenome
-        # dataset2...
-        if len(args) < 10:
+    def initialize(self, stdArgsBundle, typeSpecificArgsList):
+
+        super().initialize(stdArgsBundle, TYPE, VERSION)
+
+        if len(typeSpecificArgsList) < 4:
             raise EupathExporter.ValidationException("The tool was passed an insufficient numbers of arguments.")
 
+        if (len(typeSpecificArgsList) - 1) % 3 != 0:
+            raise EupathExporter.ValidationException("Invalid number of arguments.  Must be a reference genome followed by one or more 3-tuples.")
+
+        # Override the dataset genome reference with that provided via the form.
+        if len(typeSpecificArgsList[0].strip()) == 0:
+            raise EupathExporter.ValidationException("A reference genome must be selected.")
+        self._refGenome = ReferenceGenome.Genome(typeSpecificArgsList[0])
+
         # list arguments (for debuging)
         # print >> sys.stderr, "args to BigwigFilesEupathExporter.py"
         # for i in range(0, len(args)):
         #     print >> sys.stderr, str(args[i])
 
 
-        # grab first dataset provided ref genome
-        self._initial_refGenome = args[9]
-
         self._datasetInfos = []
         
         # process variable number of [dataset refgenome] pairs.
         # confirm that all dataset provided ref genomes are identical.
-        for i in range(7, len(args), 3):   # start on 8th arg, increment by 3
+        for i in range(1, len(typeSpecificArgsList), 3):    # start after ref genome arg, increment by tuple size (3)
+
+            path = typeSpecificArgsList[i+0]
+            filename = typeSpecificArgsList[i+1]  # user's original file name
+            rg = typeSpecificArgsList[i+2]
 
             # check file suffix (and set if needed)
-            if args[i+1].endswith(".bigwig"):
-                args[i+1] = args[i+1][0:-6] + "bw"
-            elif not args[i+1].endswith(".bw"):
-                args[i+1] = args[i+1] + ".bw"
+            if filename.endswith(".bigwig"):
+                filename = filename[0:-6] + "bw"
+            elif not filename.endswith(".bw"):
+                filename = filename + ".bw"
 
             # check file size
-            size = os.stat(args[i]).st_size
+            size = os.stat(filename).st_size
```

Comment:
Should this be `filename` or `path`?  If it's supposed to be `filename` wont changing the variable value above break the reference?